File: Numbers And Math/1304. Find N Unique Integers Sum up to Zero.cpp
class Solution {
public:
    vector<int> sumZero(int n) {
        vector<int> result;

        // If n is odd, include 0 to make the total sum zero
        if (n % 2 != 0) {
            result.push_back(0);
        }

        // Add pairs of positive and negative integers
        for (int num = 1; num <= n; num++) {
            // Stop if we've reached the required count
            if (result.size() == n) break;

            result.push_back(num);       // Add positive number
            result.push_back(-num);      // Add corresponding negative number
        }

        return result;
    }
};

File: Numbers And Math/1304. Find N Unique Integers Sum up to Zero.java
class Solution {
    public int[] sumZero(int n) {
        // Create an array of size n
        int[] result = new int[n];
        
        // Number of positive-negative pairs we can form
        int pairCount = n / 2;
        
        // Index pointer for inserting values
        int index = 0;
        
        // Fill the array with pairs like (1, -1), (2, -2), ...
        for (int i = 1; i <= pairCount; i++) {
            result[index] = i;       // insert positive number
            index++;
            result[index] = -i;      // insert negative number
            index++;
        }
        
        // If n is odd, one element will remain as 0 (default in Java)
        return result;
    }
}
